fuse-ts.c

```diff
@@ -315,6 +315,7 @@ int ts_data_do_read (sourcefile_t * file, char *buf, size_t size, off_t fileoffs
 	//debug_printf("trying to give out %d bytes at position %" PRId64 " of file '%s'\n", size, fileoffset, file->filename);
 	if (file->fhandle == NULL) {
 		file->fhandle = fopen (file->filename, "r");
+		pthread_mutex_init(&file->filelock, NULL);
```

Comment:
This should only run when the corresponding source file is being opened for the first time (if fhandle == NULL).

Reply:
You are right... but that allows another race condition on that if, I guess. Since the lock is used for every read, I think it is better to init it right along the rest of the struct, and destruct it accordingly.